AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/AWSDataStorePlugin+DataStoreSubscribeBehavior.swift

```diff
@@ -63,4 +61,25 @@ extension AWSDataStorePlugin: DataStoreSubscribeBehavior {
         operationQueue.addOperation(operation)
         return operation.publisher
     }
+    
+    public func observeQuery<M: Model>(for modelType: M.Type,
+                                       where predicate: QueryPredicate?,
+                                       sort sortInput: QuerySortInput?) async -> AmplifyAsyncThrowingSequence<DataStoreQuerySnapshot<M>> {
+        let modelSchema = modelType.schema
+        guard let dispatchedModelSyncedEvent = dispatchedModelSyncedEvents[modelSchema.name] else {
+            return Fatal.preconditionFailure("`dispatchedModelSyncedEvent` is expected to exist for \(modelSchema.name)")
+        }
+        let operation = AWSDataStoreObserveQueryOperation(modelType: modelType,
+                                                          modelSchema: modelSchema,
+                                                          predicate: predicate,
+                                                          sortInput: sortInput?.asSortDescriptors(),
+                                                          storageEngine: storageEngine,
+                                                          dataStorePublisher: dataStorePublisher,
+                                                          dataStoreConfiguration: dataStoreConfiguration,
+                                                          dispatchedModelSyncedEvent: dispatchedModelSyncedEvent)
+        operationQueue.addOperation(operation)
+        let task = AmplifyInProcessReportingOperationTaskAdapter(operation: operation)
+        return Fatal.preconditionFailure("Need to return AmplifyAsyncThrowingSequence")
+        // return task.inProcess
```

Comment:
Here `AWSDataStoreObserveQueryOperation` is created and passed to the current task adapter, which returns `AmplifyAsyncSequence`. It doesn't support returning an `AmplifyAsyncThrowingSequence`